### DISCUSSION TITLE
Exclude LazyBarrierStrategy by default

### DISCUSSION THREAD

**Discussion by porunov** [JanusGraph/janusgraph#3786] [General]
`LazyBarrierStrategy` adds `NoOpBarrierStep` with the fixed size of `2500` after `FlatMapStep`. Unless I'm missing something this shouldn't improve JanusGraph query performance anyhow. Should we just exclude this strategy by default to make `query.limited-batch-size` the only configuration needed to control default added `barrier` step size?
@rngcntr do you have any opinion about it?

**Comment by rngcntr** [comment]
In many cases, I can imagine `barrier(2500)` is not what the user wants, which leads to unwanted performance degradation. In the past, I have often used explicit `barrier(1)` steps to avoid `LazyBarrierStrategy` in places where I don't need it. However, I'm not sure there aren't any cases we are missing right now.

**Reply by porunov** [reply]
I’m not sure either but I couldn’t figure out any use-case when ‘barrier’ step after FlatMapStep improves performance anyhow. 
As far as I understand it only degrades performance a little bit with the additional barrier step being processed for no any benefit.
That said, I could be wrong and just miss some none-obvious use-cases.

**Reply by porunov** [reply]
I have created a topic in TinkerPop Discord channel ([here](https://discord.com/channels/838910279550238720/1111636664595521549)). Unless anyone provides any use-case when `LazyBarrierStrategy` has benefit when it's used with JanusGraph, I propose to disable this strategy by default.
